nomad/state/state_store.go

```diff
@@ -3160,11 +3160,12 @@ func (s *StateStore) nestedUpsertEval(txn *txn, index uint64, eval *structs.Eval
 		}
 
 		// Go through and update the evals
-		for _, eval := range blocked {
-			newEval := eval.Copy()
+		for _, blockedEval := range blocked {
+			newEval := blockedEval.Copy()
 			newEval.Status = structs.EvalStatusCancelled
-			newEval.StatusDescription = fmt.Sprintf("evaluation %q successful", newEval.ID)
+			newEval.StatusDescription = fmt.Sprintf("evaluation %q successful", eval.ID)
 			newEval.ModifyIndex = index
+			newEval.ModifyTime = eval.CreateTime
```

Comment:
The Evaluation from the `eval` argument we're upserting is `EvalStatusComplete`, so I'd expect the `eval.ModifyTime` to be updated to be later than the `eval.CreateTime` by this point. And we're updating the `ModifyIndex` to the same `ModifyIndex` as the `eval` argument (as I'd expect). So shouldn't this be?:

```suggestion
			newEval.ModifyTime = eval.ModifyTime
```

Reply:
I was convinced that this is what I initially did. Nice catch and sorry for crossing my wires :)